libexec/python/docker/tasks.py

```diff
@@ -261,3 +262,48 @@ def get_configs(manifest, keys, delim=None):
     for key in keys:
         configs[key] = get_config(manifest, key, delim=delim)
     return configs
+
+
+def create_bind_tar():
+
+    '''create_bind_tar creates a tar containing empty directories for the
+    user's $HOME and any system-wide "bind path" entries in singularity.conf
+    This allows non-overlay systems to bind these into docker containers
+    when those folders do not already exist in the container.
+    '''
+    tar_file = None
+    files = []
+
+    # Deal with home directory
+    home_bind = os.path.join(os.path.expanduser("~"),
+                             '.singularity-bind')
+    home_bind = home_bind.lstrip('/')
+    bot.verbose3('Adding home bind point to bind tar')
+    template = get_template('tarinfo')
+    template['name'] = home_bind
+    template['content'] = ("This is a bind point injected by singularity with "
```

Comment:
the content portion of the template is intended for content of files - so here you would be creating a file with name matched to `home_bind` and texty content (and not a directory). Is this what is wanted? I would think a directory would be wanted? I think you would use the equivalent template, but do an extra change to the tarinfo to make it directory type, e.g.

```
tarinfo.type = tarfile.DIRTYPE
tar.addfile(t)
```
I think adding as a file would work, but it doesn't seem exactly like what is wanted.

Reply:
So probably a check would be needed here for having content, period https://github.com/dctrud/singularity/blob/master/libexec/python/sutils.py#L180 and then another here for a type https://github.com/dctrud/singularity/blob/master/libexec/python/sutils.py#L169

Reply:
It's intentional that I'm creating a file here at . If I need bind point `/home2/dtrudgian` I'm creating the file `/home2/dtrudgian/.singularity-bind` with content that explains what that is. I wanted some kind of marker so that if I extract/inspect the image I can see the paths that were added/acted on by the inection code. When the container is run and bind is made at `/home2/` dtrudgian that file is masked as my real home dir is mounted over it.